src/indra_cogex/sources/wikidata/__init__.py

```diff
@@ -0,0 +1,431 @@
+    WIKIDATA_ENDPOINT = \
+        "https://query.wikidata.org/bigdata/namespace/wdq/sparql"
+
+    def __init_subclass__(cls, **kwargs):
+        # Modify the module path to include the name of this processor
+        cls.module = cls.module.module(cls.name)
+
+        # Now update paths for the node and relation data
+        cls.directory = cls.module.base
+        # These are nodes directly in the neo4j encoding
+        cls.nodes_path = cls.module.join(name="nodes.tsv.gz")
+        # These are nodes in the original INDRA-oriented representation
+        # needed for assembly
+        cls.nodes_indra_path = cls.module.join(name="nodes.pkl")
+        cls.edges_path = cls.module.join(name="edges.tsv.gz")
+
+    def get_nodes(self) -> Iterable[Node]:
+        raise NotImplementedError(
+            f"get_nodes not implemented in {self.__class__.__name__}. "
+            f"Implement in a subclass."
+        )
+
+    def get_relations(self) -> Iterable[Relation]:
+        raise NotImplementedError(
+            f"get_relations not implemented in {self.__class__.__name__}. "
+            f"Implement in a subclass."
+        )
+
+    def run_sparql_query(self, sparql: str) -> List[Mapping[str, Any]]:
+        logger.debug("running query: %s", sparql)
+        res = requests.get(self.WIKIDATA_ENDPOINT,
+                           params={"query": sparql, "format": "json"})
+        res.raise_for_status()
+        res_json = res.json()
+        return res_json["results"]["bindings"]
+
+
+class JournalPublisherProcessor(WikiDataProcessor):
+    """Processor for the Journal Publisher relations"""
+    name = "journal_publisher"
+    journal_node_type = "Journal"
+    publisher_node_type = "Publisher"
+    node_types = [publisher_node_type, journal_node_type]
+    sparql_query = dedent("""
+    SELECT DISTINCT
+        ?journal
+        ?journalLabel
+        (group_concat(?issn;separator=",") as ?issn_list)
+        ?issn_l
+        ?publisher
+        ?publisherLabel
+        ?isni
+    WHERE  
+    {
+        ?journal wdt:P31 wd:Q5633421 ;
+                 rdfs:label ?journalLabel ;
+                 wdt:P123 ?publisher .
+        FILTER ( LANG(?journalLabel) = "en" )
+        OPTIONAL { ?journal wdt:P236 ?issn }
+        OPTIONAL { ?journal wdt:P7363 ?issn_l }
+        OPTIONAL { ?publisher wdt:P213 ?isni }
+        SERVICE wikibase:label {
+            bd:serviceParam wikibase:language "[AUTO_LANGUAGE],en".
+        } # Helps get the label in your language, if not, then en language
+    }
+    GROUP BY ?journal ?journalLabel ?issn_l ?publisher ?publisherLabel ?isni
+    """)
+
+    def __init__(self):
+        self.publisher_data_path = self.module.join(name="publisher_data.tsv.gz")
+        self.journal_data_path = self.module.join(name="journal_data.tsv.gz")
+        self.pub_jour_relations_data_path = self.module.join(
+            name="pub_jour_relations_data.tsv.gz")
+        self.scopus_citescore_path = self.module.join(
```

Comment:
can you document where this comes from and how to get it

Reply:
Resolved in https://github.com/bgyori/indra_cogex/pull/139/commits/9add0085a5b258b79d971b04b2e06c6b7f6a8159